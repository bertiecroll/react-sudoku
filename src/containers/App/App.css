.App {}

.App-content {
  margin-top: 3em;
  display: flex;
  justify-content: center;
}

.App-logo {
  animation: App-logo-spin infinite 30s linear;
  height: 60px;
}

.App-header {
  display: flex;
  font-variant: small-caps;
  background-color: #222;
  padding: 1em;
  color: white;
}

.App-title {
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
