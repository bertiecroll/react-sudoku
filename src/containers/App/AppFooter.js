import React from 'react'

export function AppFooter () {
  return (
    <footer className="AppFooter">
      <a>Create React App</a>
      <a>Redux</a>
      <a>React Router</a>
    </footer>
  )
}
