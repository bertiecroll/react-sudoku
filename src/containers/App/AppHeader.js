import React from 'react'
import logo from 'images/logo.svg'

export function AppHeader () {
  return (
    <header className="AppHeader">
      <img src={logo} className="App-logo" alt="logo" />
      <h2 className="App-title">React Sudoku</h2>
    </header>
  )
}
