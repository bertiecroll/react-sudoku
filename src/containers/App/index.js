export { App } from './App'
export { AppHeader } from './AppHeader'
export { AppFooter } from './AppFooter'
export { default } from './App'
