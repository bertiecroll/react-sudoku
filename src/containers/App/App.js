import React from 'react'
import {
  BrowserRouter as Router,
  Route,
  Switch,
} from 'react-router-dom'
import { connect } from 'react-redux'
import './App.css';

import { AppHeader, AppFooter } from 'containers/App'
import Puzzle from 'components/Puzzle'
import Menu from 'components/Menu'
import NotFound from 'components/NotFound'

export function App () {
  return (
      <Router>
        <div className="App">
          <AppHeader />
          <div className="AppContent">
            <Switch>
              <Route exact path="/" component={Menu} />
              <Route path="/puzzle/:difficultyLevel(test|easy|medium|hard)" component={Puzzle} />
              <Route component={NotFound} />
            </Switch>
          </div>
          <AppFooter />
        </div>
      </Router>
  )
}

const mapStateToProps = state => ({})

export default connect(mapStateToProps)(App)
