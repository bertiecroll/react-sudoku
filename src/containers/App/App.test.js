import React from 'react';
import { shallow } from 'enzyme'
import { App } from './App';

describe('App', () => {
  it('renders header, footer and switch components', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })
})
