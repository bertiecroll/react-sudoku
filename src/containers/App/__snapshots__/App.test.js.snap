// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders header, footer and switch components 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <AppHeader />
    <div
      className="AppContent"
    >
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/puzzle/:difficultyLevel(test|easy|medium|hard)"
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </div>
    <AppFooter />
  </div>
</BrowserRouter>
`;
